lib/ansible/modules/network/cnos/cnos_restapi.py

```diff
@@ -0,0 +1,275 @@
+
+short_description:  Access to  RESTAPI's on switches running Lenovo CNOS.
+
+version_added: "2.5"
+
+description:
+  - "Provides access to the RESTAPI's on the Lenovo CNOS switches.
+     RESTAPI interface on CNOS fetches data and configures the
+     Lenovo CNOS switches. "
+
+options:
+    transport:
+        description:
+            - Transport layer used by the RESTAPI
+               - http  plaintext communication over port 8090
+               - https secured encrypted comminication
+        required: true
+        default: Null
+    urlpath:
+        description:
+            - URL Path of the RESTAPI
+        required: true
+        default: Null
+    method:
+        description:
+            - The HTTP method of the request
+                 - GET is typically used for querying objects
+                 - POST is typically used for creating/querying objects
+                 - PUT is typically used for modifying objects
+        required: true
+        default: Null
+    jsoninp:
+        description:
+            - input json dictionary
+                 - Used by POST, PUT method to input request paramters
+        required: false
+        default: Null
+author:
+    - Arun Kumar (@arunktele)
+'''
+
+EXAMPLES = '''
+
+- name: Configure BST feature using a JSON string
+  cnos_restapi:
+    host: '{{ inventory_hostname }}'
+    username: '{{ username }}'
+    password: '{{ password }}'
+    outputfile: "./results/test_restapi_{{ inventory_hostname }}_output.txt"
+    transport: https
+    urlpath: /nos/api/cfg/telemetry/bst/feature
+    method: PUT
+    jsoninp: '{"collection-interval": 20, "send-async-reports": 1,
+             "send-snapshot-on-trigger": 1, "trigger-rate-limit": 1,
+              "async-full-report": 0, "trigger-rate-limit-interval": 11,
+              "bst-enable": 1}'
+
+- name: Fetch BST feature using a JSON string
+  cnos_restapi:
+    host: '{{ inventory_hostname }}'
+    username: '{{ username }}'
+    password: '{{ password }}'
+    outputfile: "./results/test_restapi_{{ inventory_hostname }}_output.txt"
+    transport: https
+    urlpath: /nos/api/cfg/telemetry/bst/feature
+    method: GET
+
+- name: Fetch BST feature using a JSON string
+  cnos_restapi:
+    host: '{{ inventory_hostname }}'
+    username: '{{ username }}'
+    password: '{{ password }}'
+    outputfile: "./results/test_restapi_{{ inventory_hostname }}_output.txt"
+    transport: https
+    urlpath: /nos/api/info/telemetry/bst/congestion-drop-counters
+    method: POST
+    jsoninp: '{"req-id" : 1, "request-type" : "port-drops", "request-params": {"interface-list": ["Ethernet1/1", "Ethernet1/2", "Ethernet1/3"]}}'
+
+'''
+
+RETURN = '''
+msg:
+  description: Success or failure message
+  returned: always
+  type: string
+  sample: "RESTAPI PUT /nos/api/cfg/telemetry/bst/feature is successful"
+'''
+
+import json
+import ast
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import open_url
```

Comment:
Please use `get_url` which will give you built in error-checking

Reply:
The cnos restapi requires handling 401 error. Unfortunately the url library supresses this type of error. I had experimented with fetch_url and figured that it cannot be used, get_url cannot be used for this purpose to my knowledge. So I cannot fix this,